### DISCUSSION TITLE
Connection to SMG-5410

### DISCUSSION THREAD

**Discussion by Snited** [mbuchber/ha_heliotherm#7] [Q&A]
Hi, i am trying to connect the heatpump to the SMG-5410 Modbus gateway and the heatpump allways says "searching..." in Modem-Status.
Before i hat connected a MOXA NPort Ethernet Modem. In the heatpump the Port was configured as "Modem". I tried as "Modem" and as "direct" in the heatpump. All parameters of Serial Port are the same as in the MOXA NPort (but this has an operation mode called "ethernet modem").
Can someone tell me what i have to configure in the heatpump for modem settings and also in the modbus gateway?

thanks a lot
Jan

**Comment by mbuchber** [comment]
Did you try different speed settings in Modbus RTU settings?

**Reply by Snited** [reply]
i tried many diffferent things. i can see in the console that the gateway talks to the heatpump but the heatpumt does not answer. if i set the modem in the heatpump from direct to modem i can see the heatpump talking and waiting for AT commands in the console. so i think the gateway does not start the communication to the heatpump properly...

**Comment by micbac3012** [comment]
Did you get it working? Struggling the Same issue and looking for Somebody who got IT working with a SMG-5410  ...

**Reply by Snited** [reply]
sadly no

**Reply by Neumann12** [reply]
Was anyone successful by implementation of SMG-5410?

**Comment by kobius77** [comment]
honestly i doubt this can work - without additional software.

probably this software might run on such converters, but then one would have to manually configure this to map the datapoints (read/writeable via the serial port) to modbus register numbers - exactly the ones the Heliotherm RCG uses, as ha_heliotherm is programmed for these.

 so if it works - it's not plug'n'play - besides the connection issues you have.

the actual (working) solution to replace the RCG would be this: https://github.com/dstrigl/HtREST , but afaik there's no HA integration using this (though someone recently built [a webinterface](https://github.com/agilastic/htrest-monitor) ). But one could use the RESTful integrations of HA for this. Quite some configuration work to do either, but maybe there are configs to find yet.
